### ISSUE TITLE
[Accessibility]: Announce dynamic form elements

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35188]
### Audit type

Screen reader navigation

### User journey audit issue

#33516, #33518

### Problem description

<img width="884" alt="stepper" src="https://user-images.githubusercontent.com/103087/167504298-2f9bc469-f58e-4e9a-91b7-223195bebf28.png">

Selecting "custom" on the duration and max age fields of the code intelligence policy details page will re-read the already selected control.

### Expected behavior

It should put focus on the new element immediately.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by macraig** [comment]
Closing as identified issues have been completed
